index.js

```diff
@@ -123,6 +127,13 @@ function buildRules(ruleTuples) {
   });
 }
 
+function lookFor(userAgentString, leadingTokens, trailingTokens) {
```

Comment:
Not sure I'm happy with this function name.  Might be better to rename it to `lookForInUA`.

Reply:
The function name should specify exactly what it does and what it's used for.

In this case, maybe `deviceTypeExtraction`?